Content.Server/Charges/Systems/ChargesSystem.cs

```diff
@@ -0,0 +1,63 @@
+using Content.Server.Charges.Components;
+using Content.Shared.Charges.Components;
+using Content.Shared.Charges.Systems;
+using Content.Shared.Examine;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Charges.Systems;
+
+public sealed class ChargesSystem : SharedChargesSystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AutoRechargeComponent, EntityUnpausedEvent>(OnUnpaused);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var (charges, recharge) in EntityQuery<LimitedChargesComponent, AutoRechargeComponent>())
```

Comment:
This should use `EntityQueryEnumerator` so you can get uid without using .Owner

Reply:
oh that exists real